Functions/QualityControl/xASL_qc_GenerateReport.m

```diff
@@ -0,0 +1,418 @@
+    
+    if isfield(input, 'overlay')
+        fields = fieldnames(input.overlay);
+        for iField=1:length(fields)
+            if isfield(x.P, fields(iField))
+                ImIn(end+1) = {x.P.(fields{iField})};
+                header = [header ' + ' fields{iField}];
+            end
+        end
+    end
+
+    if isfield(input, 'slice')
+        [x.S.TraSlices] = xASL_sub_getSliceFromStruct(input.slice, 'TraSlice');
+        [x.S.CorSlices] = xASL_sub_getSliceFromStruct(input.slice, 'CorSlice');
+        [x.S.SagSlices] = xASL_sub_getSliceFromStruct(input.slice, 'SagSlice');       
+    else
+        x.S.TraSlices = [25, 50, 90];
+        x.S.SagSlices = [25, 50, 90];
+        x.S.CorSlices = [25, 50, 90];
+    end
+
+    img = xASL_vis_CreateVisualFig(x, ImIn, [], [], [], [], [], [], [], [], [], [], []);
+    fg = imshow(img);
+    clear fg
+    settings.figurecount = xASL_sub_PrintHeader(header, figure, settings, canvas);
+end
+
+function [slice] = xASL_sub_getSliceFromStruct(struct, name)
+
+    if isfield(struct, name) && ~isempty(struct.(name))
+        slice = str2num(struct.(name));
+    else 
+        slice = [];
+    end
+end
+
+function [newCanvas, line] = xASL_sub_createNewCanvas(position, size, oldCanvas)
+    newPosition = [oldCanvas(1) + position(1) * oldCanvas(3) oldCanvas(2) + position(2) * oldCanvas(4)];
+    newSize = [oldCanvas(3:4) .* size];
+    newCanvas = [newPosition, newSize];
+    line = [newPosition(1)  newPosition(2) + newSize(2) newSize(1) 0];
+end
+
+function line = xASL_sub_PrintText(input, figure, line, settings)
+
+    switch class(input)
+        case {'string', 'char'} 
+            String = input;
+        case 'struct'
+            String = input.text;
+            if isfield(input, 'settings')
+                settings = xASL_sub_loadSettings(input.settings, settings);
+            end
+        otherwise
+            class(input)
+            error('xASL_sub_PrintText couldnt find string of printable text')
+    end
+
+    ax=axes('Position', line ,'Visible', settings.axesVisible, 'Parent', figure);
+    text(0,0, String, 'Parent', ax, 'FontSize', settings.fontSize,'FontWeight', settings.fontWeight, 'Color', settings.color, 'FontName', settings.fontName, 'Interpreter', 'none', 'VerticalAlignment','top');
+    line = xASL_sub_NewLine(line, settings);
+end
+
+function line = xASL_sub_NewLine(line, settings)
+    line(2) = line(2) - (settings.lineSpacing) - (settings.fontSize * 0.001);
+
+    if line(2) < 0 
+        warning('No space left on page!');
+    elseif line(2) < settings.canvas(2) 
+        warning('Printing outside canvas, check block settings.');
+    end
+end
+
+function [figurecount] = xASL_sub_PrintHeader(header, figure, settings, position)
+
+    if isempty(header)
+        figurecount = settings.figurecount;
+        return
+    end
+
+    position(4) = 0;
+    settings.HorizontalAlignment = 'center';
+    figurecount = settings.figurecount + 1;
```

Comment:
why is figureCount (check camelCase) defined twice, what is the role of "settings"? Make it readable